File: spark/geospark/src/main/java/com/examples/hops/spark/geospark/HDFSWriter.java
package com.examples.hops.spark.geospark;

import org.apache.hadoop.conf.Configuration;
import org.apache.hadoop.fs.FSDataOutputStream;
import org.apache.hadoop.fs.FileSystem;
import org.apache.hadoop.fs.LocalFileSystem;
import org.apache.hadoop.fs.Path;
import org.apache.hadoop.hdfs.DistributedFileSystem;

import java.io.IOException;
import java.util.logging.Logger;


public class HDFSWriter implements AutoCloseable {
    private static Logger  log = Logger.getLogger(HDFSWriter.class.getName());
    private FSDataOutputStream outputStream;

    public HDFSWriter(String path) throws IOException {

        //Create a path
        Path hdfswritepath = new Path(path);
        Configuration conf = new Configuration();
        conf.set("fs.hdfs.impl", DistributedFileSystem.class.getName());
        conf.set("fs.file.impl", LocalFileSystem.class.getName());
        FileSystem fs = FileSystem.get(conf);

        //Init output stream
        outputStream  = fs.create(hdfswritepath);
    }

    @Override
    public void close() throws IOException {
        outputStream.close();
    }



  public void write(String comments, Boolean index, Long count, Double avgDuration, Double warmupDuration, Integer n_iterations, Integer n_partitions, Integer n_cores) throws IOException {
    log.info("Begin Write file into hdfs");
    //Classical output stream usage
    outputStream.writeBytes("comments, index, partitions (per dimension), cores, iterations, warmup time, avg execution time, resultset count\n");
    outputStream.writeBytes(comments + ", " + index + ", " + n_partitions + ", " + n_cores + ", " + n_iterations + ", " + warmupDuration + ", " + avgDuration + ", " + count +"\n");
    log.info("End Write file into hdfs");
  }

}

File: spark/stark/src/main/scala/com/examples/hops/spark/utils/HDFSWriter.scala
package com.examples.hops.spark.utils

import java.util.logging.Logger

import org.apache.hadoop.conf.Configuration
import org.apache.hadoop.fs.{FSDataOutputStream, FileSystem, LocalFileSystem, Path}
import org.apache.hadoop.hdfs.DistributedFileSystem


class HDFSWriter(val path: String) {
  val log: Logger = Logger.getLogger(classOf[HDFSWriter].getName)

  //Create a path
  val hdfswritepath = new Path(path)
  val configuration = new Configuration()
  configuration.set("fs.hdfs.impl", classOf[DistributedFileSystem].getName)
  configuration.set("fs.file.impl", classOf[LocalFileSystem].getName)
  val fs: FileSystem = FileSystem.get(configuration)

  //Init output stream
  val outputStream: FSDataOutputStream = fs.create(hdfswritepath)

  def write(comments: String, index: Boolean, count: Long, avgDuration: Double, warmupDuration: Double, n_iterations: Int, n_partitions: Int, n_cores: Int): Unit ={
    log.info("Begin Write file into hdfs")
    //Classical output stream usage
    outputStream.writeBytes("comments, index, partitions (per dimension), cores, iterations, warmup time, avg execution time, resultset count\n")
    outputStream.writeBytes(comments + ", " + index + ", " + n_partitions + ", " + n_cores + ", " + n_iterations + ", " + warmupDuration + ", " + avgDuration + ", " + count +"\n")
    log.info("End Write file into hdfs")
  }

  def close(): Unit ={
    outputStream.close()
  }

}

